src/realm/sync/client_base.hpp

```diff
@@ -245,14 +246,20 @@ struct ClientConfig {
 ///
 /// \sa set_connection_state_change_listener().
 struct SessionErrorInfo : public ProtocolErrorInfo {
+    SessionErrorInfo(const ProtocolErrorInfo& info)
+        : ProtocolErrorInfo(info)
+        , status(protocol_error_to_status(static_cast<ProtocolError>(info.raw_error_code), message))
+    {
+    }
+
     SessionErrorInfo(const ProtocolErrorInfo& info, Status status)
         : ProtocolErrorInfo(info)
         , status(std::move(status))
     {
     }
 
     SessionErrorInfo(Status status, IsFatal is_fatal)
-        : ProtocolErrorInfo(status.get_std_error_code().value(), status.reason(), is_fatal)
+        : ProtocolErrorInfo(0, {}, is_fatal)
```

Comment:
shouldn't we pass the info from `status` to the ProtocolErrorInfo constructor?

Reply:
So this is the case where if you are constructing a SessionErrorInfo for something besides a protocol error. I wasn't sure what the right thing to do here is/was, but I settled on leaving the ProtocolErrorInfo portion of the class in an invalid state rather than trying to reverse-translate Status's to ProtocolErrors.